op-batcher/batcher/channel_manager.go

```diff
@@ -202,11 +215,15 @@ func (s *channelManager) getReadyChannel(l1Head eth.BlockID) (*channel, error) {
 	dataPending := firstWithTxData != nil
 	s.log.Debug("Requested tx data", "l1Head", l1Head, "txdata_pending", dataPending, "blocks_pending", len(s.blocks))
 
-	// Short circuit if there is pending tx data or the channel manager is closed.
-	if dataPending || s.closed {
+	// Short circuit if there is pending tx data or the channel manager is closed
+	if dataPending {
 		return firstWithTxData, nil
 	}
 
+	if s.closed {
+		return nil, io.EOF
+	}
```

Comment:
Why this new check? Why not at the top of the function?

Reply:
The existing behavior is such that it is possible to return `nil,nil` (when the channel manager is closed and when there is no channel found in the search for `firstWithTxData`). 

It is important to not return early when the channel manager is closed, in case we are trying to still finish submitting a channel.